frodo.js

```diff
@@ -9,32 +9,34 @@
  * Copyright 2016 Henrik Ingo (@henrikingo)
```

Comment:
SCRIPT-This function seems to have grown quite complex. Can we break it down further?